<?xml version="1.0" encoding="utf-8"?>
<launch>
  <arg name="host" default="localhost" />
  <arg name="port" default="8888" />
  <arg name="gui" default="false" />
  <arg name="run_api" default="false" />
  <arg name="VQA_INPUT_IMAGE" default="vqa_image" />
  <arg name="image_transport" default="raw" />

  <node name="ofa_api" pkg="jsk_perception" type="run_jsk_vil_api" output="log"
        args="ofa -p $(arg port)" if="$(arg run_api)" />

  <node name="vqa" pkg="jsk_perception" type="vqa_node.py" output="screen">
    <remap from="~image" to="$(arg VQA_INPUT_IMAGE)" />
    <rosparam subst_value="true">
      host: $(arg host)
      port: $(arg port)
      image_transport: $(arg image_transport)
    </rosparam>
  </node>

  <include file="$(find jsk_perception)/launch/ofa_gui.launch" if="$(arg gui)" />

</launch>
